src/sdi-theme-monitor.c

```diff
@@ -91,12 +92,22 @@ static void notification_closed_cb(NotifyNotification *notification,
                                    SdiThemeMonitor *self) {
   /* Notification has been closed: */
   g_clear_object(&self->install_notification);
+  self->install_notification_answered = FALSE;
 }
 
 static void notify_cb(NotifyNotification *notification, gchar *action,
                       gpointer user_data) {
   SdiThemeMonitor *self = user_data;
 
+  /* Some notification daemons, when the user presses an action, emit
+   * the action first, and after it, the "default" action. Although
+   * this is an incorrect behavior, we add a workaround to avoid more
+   * bug reports. */
+  if (self->install_notification_answered == TRUE) {
+    return;
+  }
+
+  self->install_notification_answered = TRUE;
```

Comment:
Can't you instead disconnect the signal on first callback and connect it back when needed?

Or bock it instead?

Reply:
Mmm... good point...

Reply:
Ok, it's not possible because they aren't signals, but specific callbacks in the notification, and it seems that it's not possible to "disable" them. Only "close" is a true GObject signal. I'll add a comment explaining it.